### ISSUE TITLE
Issue with most recent commit

### ISSUE THREAD

**Issue by E85Addict** [j-hc/revanced-magisk-module#510]
I think everything on the config_update function is perfectly fine, but it seems the the changes in get_rv_prebuilts have caused an issue which duplicates again, causing the "./build.sh config.toml --config-update" command in the CI to trigger a new build because of it. 

I ran the main branch of the repo unchanged and this is what the build.md puts out. 

```
Music (arm-v7a): 7.20.51  
Music (arm64-v8a): 7.20.51  
Music-Extended (arm64-v8a): 7.16.53  
Twitter: 10.60.0-release.0  
YouTube: 19.16.39  
YouTube-Extended: 19.16.39  
TikTok: 36.5.4  
Twitch: 16.9.1  

Install [Microg](https://github.com/ReVanced/GmsCore/releases) for non-root YouTube and YT Music APKs  
Use [zygisk-detach](https://github.com/j-hc/zygisk-detach) to detach root ReVanced YouTube and YT Music from Play Store  

[revanced-magisk-module](https://github.com/j-hc/revanced-magisk-module)
  
CLI: inotia00/revanced-cli-4.6.1-all.jar  
Integrations: inotia00/revanced-integrations-1.14.1.apk  
Patches: inotia00/revanced-patches-4.14.1.jar  
[Changelog](https://github.com/inotia00/revanced-patches/releases/tag/v4.14.1)

CLI: inotia00/revanced-cli-4.6.1-all.jar  
Integrations: inotia00/revanced-integrations-1.14.1.apk  
Patches: inotia00/revanced-patches-4.14.1.jar  
CLI: inotia00/revanced-cli-4.6.1-all.jar  
Integrations: inotia00/revanced-integrations-1.14.1.apk  
Patches: inotia00/revanced-patches-4.14.1.jar  
CLI: j-hc/revanced-cli-4.6.0-all.jar  
Integrations: ReVanced/revanced-integrations-1.14.2.apk  
Patches: ReVanced/revanced-patches-4.15.0.jar  
[Changelog](https://github.com/ReVanced/revanced-patches/releases/tag/v4.15.0)

CLI: j-hc/revanced-cli-4.6.0-all.jar  
Integrations: ReVanced/revanced-integrations-1.14.2.apk  
Patches: ReVanced/revanced-patches-4.15.0.jar  
CLI: j-hc/revanced-cli-4.6.0-all.jar  
Integrations: ReVanced/revanced-integrations-1.14.2.apk  
Patches: ReVanced/revanced-patches-4.15.0.jar  
CLI: j-hc/revanced-cli-4.6.0-all.jar  
Integrations: ReVanced/revanced-integrations-1.14.2.apk  
Patches: ReVanced/revanced-patches-4.15.0.jar  
CLI: j-hc/revanced-cli-4.6.0-all.jar  
Integrations: ReVanced/revanced-integrations-1.14.2.apk  
Patches: ReVanced/revanced-patches-4.15.0.jar    

```

**Comment by j-hc** [comment]
cant reproduce

**Comment by E85Addict** [comment]
> cant reproduce

I recorded myself reproducing the issue. https://www.youtube.com/watch?v=uSzmXQ9m2kI

**Comment by j-hc** [comment]
need to see it on a public repo. still cant reproduce

**Comment by E85Addict** [comment]
> need to see it on a public repo. still cant reproduce

I ran it on a personal [public repo](https://github.com/E85Addicts/revanced-magisk-module) and the issue does not happen there. I guess it is only an issue when ran locally on my machine. I'd imagine the reason for this is because locally the nproc would use all my cores and threads vs the github actions being limited. I will try using the parallel-jobs variable local to see if that would technically fix the issue.

**Comment by E85Addict** [comment]
> > need to see it on a public repo. still cant reproduce
> 
> I ran it on a personal [public repo](https://github.com/E85Addicts/revanced-magisk-module) and the issue does not happen there. I guess it is only an issue when ran locally on my machine. I'd imagine the reason for this is because locally the nproc would use all my cores and threads vs the github actions being limited. I will try using the parallel-jobs variable local to see if that would technically fix the issue.

Yeah, don't think that's it either, my just be a weird issue on my end. Locally it has the issue but not on the repo my mistake.

**Comment by j-hc** [comment]
> nproc would use all my cores and threads

revanced prebuilts are not downloaded in parallel irregardless of the parallel-jobs option. if you can debug out the issue, lmk.

**Comment by E85Addict** [comment]
> > nproc would use all my cores and threads
> 
> revanced prebuilts are not downloaded in parallel irregardless of the parallel-jobs option. if you can debug out the issue, lmk.

With the latest "fixes" commit, building locally fails inotia00 builds but fixes the duplicating. I noticed if I move this [if statement](https://github.com/j-hc/revanced-magisk-module/commit/45bf672667470aef8a9b79aa9ee19dd2efa68df7#diff-35a54395325de0227bbdd613dbf61c3265b5f6ab205cf32348e3538399cf1b9eR83-R85) to after this [line](https://github.com/j-hc/revanced-magisk-module/commit/45bf672667470aef8a9b79aa9ee19dd2efa68df7#diff-35a54395325de0227bbdd613dbf61c3265b5f6ab205cf32348e3538399cf1b9eR70) it completely fixes all issues locally.

**Comment by E85Addict** [comment]
Also tested that change on a public repo, works perfectly. Here is the run https://github.com/E85Addicts/revanced-magisk-module/actions/runs/11097032800

**Comment by E85Addict** [comment]
> > Also tested that [change](https://github.com/j-hc/revanced-magisk-module/commit/bd69b742fec6fd2d1b67ffb16d4e31e56eee9fcb)
> 
> this does not make sense. you are grepping in $file when its empty.

Understandable, it's just odd that building locally, it fixes the issue.

**Comment by E85Addict** [comment]
With the latest commits, this is the new error, and it just hangs here. 

```
e85addict@AMD-PC:~/dev/revanced-magisk-module$ ./build.sh
[+] Getting prebuilts (ReVanced)
2024-10-05 16:09:44 URL:https://api.github.com/repos/j-hc/revanced-cli/releases/latest [3684/3684] -> "-" [1]
[+] Getting '/home/e85addict/dev/revanced-magisk-module/temp/j-hc-rv/revanced-cli-4.6.0-all.jar' from 'https://api.github.com/repos/j-hc/revanced-cli/releases/assets/159960172'
2024-10-05 16:09:45 URL:https://objects.githubusercontent.com/github-production-release-asset-2e65be/531294240/527ba4ab-0de3-4b6d-a543-1e5d6b123734?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20241005%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241005T230945Z&X-Amz-Expires=300&X-Amz-Signature=ff8a53579e1679f4f4b500fd07fe1d18a69190b7089a5ab511d2f208b81fcfdf&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Drevanced-cli-4.6.0-all.jar&response-content-type=application%2Foctet-stream [50857113/50857113] -> "/home/e85addict/dev/revanced-magisk-module/temp/j-hc-rv/tmp.revanced-cli-4.6.0-all.jar" [1]
2024-10-05 16:09:46 URL:https://api.github.com/repos/ReVanced/revanced-integrations/releases/latest [6760/6760] -> "-" [1]
[+] Getting '/home/e85addict/dev/revanced-magisk-module/temp/revanced-rv/revanced-integrations-1.15.0.apk' from 'https://api.github.com/repos/ReVanced/revanced-integrations/releases/assets/195953872'
2024-10-05 16:09:48 URL:https://objects.githubusercontent.com/github-production-release-asset-2e65be/470324030/11ac79d6-ed34-45d7-9939-aeb09611e32d?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20241005%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241005T230948Z&X-Amz-Expires=300&X-Amz-Signature=83f970b34a2707ce7bc2ec435dcd35972722b328de7c5edc60de9e7b5dff66bb&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Drevanced-integrations-1.15.0.apk&response-content-type=application%2Fvnd.android.package-archive [280320/280320] -> "/home/e85addict/dev/revanced-magisk-module/temp/revanced-rv/tmp.revanced-integrations-1.15.0.apk" [1]
2024-10-05 16:09:48 URL:https://api.github.com/repos/ReVanced/revanced-patches/releases/latest [9904/9904] -> "-" [1]
[+] Getting '/home/e85addict/dev/revanced-magisk-module/temp/revanced-rv/revanced-patches-4.16.0.jar' from 'https://api.github.com/repos/ReVanced/revanced-patches/releases/assets/195950676'
2024-10-05 16:09:50 URL:https://objects.githubusercontent.com/github-production-release-asset-2e65be/731390704/7d3fa8a2-b9c4-43c5-acdc-bba7ccd51409?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20241005%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241005T230950Z&X-Amz-Expires=300&X-Amz-Signature=f53e17e1448a879975a04f67992218c7875304c3b96b8e09638f15ad20fa57ff&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Drevanced-patches-4.16.0.jar&response-content-type=application%2Foctet-stream [3185307/3185307] -> "/home/e85addict/dev/revanced-magisk-module/temp/revanced-rv/tmp.revanced-patches-4.16.0.jar" [1]
2024-10-05 16:09:51 URL:https://api.github.com/repos/ReVanced/revanced-patches/releases/latest [9904/9904] -> "-" [1]
[+] Getting '/home/e85addict/dev/revanced-magisk-module/temp/revanced-rv/patches-v4.16.0.json' from 'https://api.github.com/repos/ReVanced/revanced-patches/releases/assets/195950673'
2024-10-05 16:09:52 URL:https://objects.githubusercontent.com/github-production-release-asset-2e65be/731390704/796a24a3-5422-4b8e-b8b9-6c0f497daea5?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20241005%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241005T230952Z&X-Amz-Expires=300&X-Amz-Signature=500b414b3da3f02c3097919a8bde76bdf0bfcf315ced62f0c92c464f7935c4c4&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dpatches.json&response-content-type=application%2Foctet-stream [86519/86519] -> "/home/e85addict/dev/revanced-magisk-module/temp/revanced-rv/tmp.patches-v4.16.0.json" [1]
[+] Getting prebuilts (ReVanced)
[-] ABORT: filter fail: '/home/e85addict/dev/revanced-magisk-module/temp/j-hc-rv/revanced-cli-4.6.0-all.jar' with 'latest'
[-] ABORT: could not download rv prebuilts
::error::utils.sh [-] ABORT: could not download rv prebuilts

e85addict@AMD-PC:~/dev/revanced-magisk-module$ 2024-10-05 16:09:54 URL:https://dn790002.ca.archive.org/0/items/jhc-apks/apks/com.google.android.youtube/ [659] -> "-" [1]
[+] Choosing version '19.16.39' for YouTube
[+] Downloading 'YouTube' from archive


```

**Comment by j-hc** [comment]
fixed 01c85cb

**Comment by E85Addict** [comment]
> fixed [01c85cb](https://github.com/j-hc/revanced-magisk-module/commit/01c85cb7a2813f777bd8449ed6d9ca73183eccc4)

Amazing work, all is working well now1
